script/check_publish.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -e
+
+# This script checks to make sure that each of the plugins *could* be published.
+# It doesn't acutually publish anything.
+
+# So that users can run this script from anywhere and it will work as expected.
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+REPO_DIR="$(dirname "$SCRIPT_DIR")"
+
+function error() {
+  echo "$@" 1>&2
+}
+
+function check_publish() {
+  local failures=()
+  for package_name in "$@"; do
+    local dir="$REPO_DIR/packages/$package_name"
+    echo "Checking that $package_name can be published."
+    if (cd "$dir" && pub publish --dry-run > /dev/null); then
+      echo "Package $package_name is able to be published."
+    else
+      error "Unable to publish $package_name"
+      failures=("${failures[@]}" "$package_name")
+    fi
+  done
+  if [[ "${#failures[@]}" != 0 ]]; then
+    echo "FAIL: The following ${#failures[@]} package(s) failed the publishing check:"
+    for failure in "${failures[@]}"; do
+      echo "$failure"
+    done
+  fi
+  return "${#failures[@]}"
+}
+
+function check_changed_packages() {
```

Comment:
It seems that we repeat this function. Perhaps define a common script and include it using `source`?